tracer/src/Datadog.Trace/Ci/CIVisibility.cs

```diff
@@ -361,7 +361,14 @@ internal static IApiRequestFactory GetRequestFactory(ImmutableTracerSettings tra
 
 #if NETCOREAPP
             Log.Information("Using {FactoryType} for trace transport.", nameof(HttpClientRequestFactory));
-            factory = new HttpClientRequestFactory(tracerSettings.ExporterInternal.AgentUriInternal, AgentHttpHeaderNames.DefaultHeaders, timeout: timeout);
+            factory = new HttpClientRequestFactory(
+                tracerSettings.ExporterInternal.AgentUriInternal,
+                AgentHttpHeaderNames.DefaultHeaders,
+                handler: new System.Net.Http.HttpClientHandler
+                {
+                    AutomaticDecompression = DecompressionMethods.GZip | DecompressionMethods.Deflate,
+                },
```

Comment:
Makes me think about the fact that sockethandler is more performant IIRC - wonder if we should be using that with .NET 6+ instead of forcing httpclienthandler 🤔 Unrelated to this PR though 😅

Reply:
SocketHandler is used behind the scenes AFAIK: https://github.com/dotnet/runtime/blob/v6.0.0/src/libraries/System.Net.Http/src/System/Net/Http/HttpClientHandler.cs#L16